.github/workflows/test-zkvm.yml

```diff
@@ -164,6 +136,54 @@ jobs:
             BASE_IMAGE_TAG=${{ steps.inspect_image.outputs.base_image_tag }}
             CI=1
 
+  clippy_via_docker:
+    name: Clippy via Docker
+    needs: build_image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v2
+        with:
+          key: rust-${{ inputs.toolchain }}-${{ inputs.zkvm }}-${{ hashFiles('Cargo.lock') }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Pull image ere-base-${{ inputs.zkvm }}
+        run: docker image pull ${{ needs.build_image.outputs.base_zkvm_image_tag }}
+
+      - name: Run cargo clippy for ere-${{ inputs.zkvm }} via Docker
+        run: |
+          DOCKER_CMD="docker run \
+            --rm \
+            --interactive \
+            --volume ${{ github.workspace }}:/ere \
+            --volume $HOME/.cargo:/root/.cargo \
+            --workdir /ere \
+            ${{ needs.build_image.outputs.base_zkvm_image_tag }} \
+            /bin/bash"
+
+          cat <<EOF | $DOCKER_CMD
+          set -e
+
+          OPTIONS="--all-targets ${{ inputs.default_features && '' || '--no-default-features' }} -- -D warnings"
+          cargo clippy --package ere-${{ inputs.zkvm }} \$OPTIONS
+          cargo clippy --package ere-cli --features cli,${{ inputs.zkvm }} \$OPTIONS
+
+          chown -R $(id -u):$(id -g) ~/.cargo
+          chown -R $(id -u):$(id -g) target
```

Comment:
Because in docker the user is currently root, we need to change it back to make sure latter caching work